Tests/Tests/SDAnimatedImageTest.m

```diff
@@ -13,6 +13,41 @@
 
 static const NSUInteger kTestGIFFrameCount = 5; // local TestImage.gif loop count
 
+// Check whether the coder is called
+@interface SDImageAPNGTestCoder : SDImageAPNGCoder
+
+@property (nonatomic, class, assign) BOOL isCalled;
```

Comment:
This property can use instance level property. No need a class property.

expect(SDImageAPNGTestCoder.sharedCoder.isCalled).beTruth();

Reply:
If the property is an instance level property, we can't access it. The new coder created in `initWithAnimatedImageData` holds it, not the sharedCoder.

Reply:
OK